### ISSUE TITLE
Please verify that the package.json has a valid "main" entry 

### ISSUE THREAD

**Issue by luismadrigal** [papercss/papercss#208]
Apparently package.json is referencing "main": "index.js" but there is not "index.js" file anywhere.

This problem happens on installs via yarn and npm.

```
Module build failed (from ./node_modules/postcss-loader/src/index.js):
Error: Cannot find module '.../node_modules/papercss/index.js'. Please verify that the package.json has a valid "main" entry
```

Any workarounds?

**Comment by rhyneav** [comment]
Hey @luismadrigal sorry to hear that! Can I ask what operating system you're on and what your current `package.json` file looks like? I was just able to install into an empty codebase with no problem, so I'm curious if PaperCSS isn't playing nice with another package. 

As for workarounds, you can download the latest css files from the releases page here and use those in your application: https://github.com/papercss/papercss/releases

**Comment by luismadrigal** [comment]
Hi @rhyneav, thanks for getting back to me.

I'm on MacOS Catalina.  I'm developing a Rails (6.0.3.2) app and using yarn (1.22.4) npm (6.14.5) for installing the Node_Modules and webpack (4.44.0) complains about the invalid "main" : "index.js".  

Here is the package.json from the Rails app

```
{
  "name": "simple_blog",
  "private": true,
  "dependencies": {
    "@rails/actioncable": "^6.0.0",
    "@rails/activestorage": "^6.0.0",
    "@rails/ujs": "^6.0.0",
    "@rails/webpacker": "4.2.2",
    "turbolinks": "^5.2.0"
  },
  "version": "0.1.0",
  "devDependencies": {
    "webpack-dev-server": "^3.11.0"
  }
}
```

Let me know if you need anything else.

Thanks.

P.S.  I've downloaded papercss locally so I can start developing, I'm just hoping that at some point it will be available as a package/module.

**Comment by luismadrigal** [comment]
Hi @rhyneav, I'm realizing that the package.json you were asking about is the one for the papercss package, so I'm adding it here below.  I believe the line in question is line 5 **"main": "index.js"**  

```
{
  "name": "papercss",
  "version": "1.7.0",
  "description": "The less formal CSS framework.",
  "main": "index.js",                     
  "scripts": {
    "start": "node node_modules/gulp/bin/gulp.js",
    "build": "node node_modules/gulp/bin/gulp.js build",
    "test": "echo \"Error: no test specified\" && exit 1",
    "postinstall": "gulp postinstall",
    "stylelint": "node node_modules/stylelint/bin/stylelint.js 'src/**/*.scss'"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/papercss/papercss.git"
  },
  "author": "Rhyne Vlaservich <rhyneav@gmail.com>",
  "license": "ISC",
  "bugs": {
    "url": "https://github.com/papercss/papercss/issues"
  },
  "homepage": "https://www.getpapercss.com",
  "dependencies": {
    "gulp": "^4.0.0",
    "gulp-autoprefixer": "^6.0.0",
    "gulp-clean-css": "3.9.0",
    "gulp-connect": "^5.6.1",
    "gulp-plumber": "1.1.0",
    "gulp-rename": "1.2.2",
    "gulp-sass": "^3.1.0",
    "optional": "^0.1.4"
  },
  "devDependencies": {
    "browserify": "^14.1.0",
    "chai": "^4.1.2",
    "gulp-less": "^4.0.1",
    "gulp-plumber": "1.1.0",
    "gulp-rename": "1.2.2",
    "gulp-sass": "^3.1.0",
    "gulp-stylelint": "6.0.0",
    "gulp-watch-less": "1.0.1",
    "http-server": "^0.11.1",
    "hugo-bin": "^0.37.0",
    "install": "0.10.1",
    "jake": "^8.0.15",
    "jshint": "^2.9.6",
    "karma": "^3.1.1",
    "karma-chai": "^0.1.0",
    "karma-commonjs": "1.0.0",
    "karma-firefox-launcher": "^1.0.0",
    "karma-mocha": "^1.3.0",
    "karma-node-modules-middleware": "^1.0.1",
    "karma-quixote": "^1.0.0",
    "karma-requirejs": "^1.1.0",
    "mocha": "^4.0.1",
    "nodemon": "^1.11.0",
    "npm": "^6.4.1",
    "object-merge": "^2.5.1",
    "os": "^0.1.1",
    "pre-commit": "^1.2.2",
    "procfile": "^0.1.1",
    "quixote": "^0.14.0",
    "request": "^2.83.0",
    "requirejs": "^2.3.5",
    "semver": "^5.3.0",
    "shelljs": "^0.7.6",
    "simplebuild-jshint": "^1.3.0",
    "simplebuild-karma": "^1.0.0",
    "stylelint": "8.4.0",
    "stylelint-config-sass-guidelines": "4.0.1",
    "stylelint-config-standard": "18.0.0",
    "stylelint-order": "0.8.0",
    "stylelint-scss": "2.2.0",
    "webpack-dev-server": "^3.1.10"
  },
  "pre-commit": [
    "stylelint"
  ]
}
```

I hope this is more helpful.

Thanks

**Comment by rhyneav** [comment]
Both are super helpful, thank you for the information! I'll look into this further.

**Comment by rhyneav** [comment]
Hey @luismadrigal I'm not able to reproduce this on my system. Could you please send the full install command you're running as well as the output? This is what it looks like for me:

With Yarn
```
> yarn add papercss
yarn add v1.22.4
[1/4] 🔍  Resolving packages...
warning papercss > gulp > glob-watcher > chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
warning papercss > gulp-clean-css > gulp-util@3.0.8: gulp-util is deprecated - replace it, following the guidelines at https://medium.com/gulpjs/gulp-util-ca3b1f9f9ac5
warning papercss > gulp-sass > gulp-util@3.0.8: gulp-util is deprecated - replace it, following the guidelines at https://medium.com/gulpjs/gulp-util-ca3b1f9f9ac5
warning papercss > gulp-plumber > gulp-util@3.0.8: gulp-util is deprecated - replace it, following the guidelines at https://medium.com/gulpjs/gulp-util-ca3b1f9f9ac5
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
warning " > webpack-dev-server@3.11.0" has unmet peer dependency "webpack@^4.0.0 || ^5.0.0".
warning "webpack-dev-server > webpack-dev-middleware@3.7.2" has unmet peer dependency "webpack@^4.0.0".
[4/4] 🔨  Building fresh packages...
success Saved lockfile.
success Saved 121 new dependencies.
info Direct dependencies
└─ papercss@1.7.0
info All dependencies
├─ ansi-gray@0.1.1
├─ ansi-wrap@0.1.0
├─ append-buffer@1.0.2
├─ archy@1.0.0
├─ arr-filter@1.1.2
├─ arr-map@2.0.2
├─ array-differ@1.0.0
├─ array-each@1.0.1
├─ array-initial@1.1.0
├─ array-last@1.3.0
├─ array-sort@1.0.0
├─ async-settle@1.0.0
├─ bach@1.2.0
├─ beeper@1.1.1
├─ body@5.1.0
├─ buffer-equal@1.0.0
├─ clean-css@4.1.9
├─ clone-buffer@1.0.0
├─ clone-stats@1.0.0
├─ clone@2.1.2
├─ cloneable-readable@1.1.3
├─ collection-map@1.0.0
├─ connect-livereload@0.6.1
├─ connect@3.7.0
├─ continuable-cache@0.3.1
├─ copy-props@2.0.4
├─ d@1.0.1
├─ dateformat@2.2.0
├─ default-compare@1.0.0
├─ default-resolution@2.0.0
├─ duplexer2@0.0.2
├─ each-props@1.3.2
├─ error@7.2.1
├─ es5-ext@0.10.53
├─ es6-iterator@2.0.3
├─ ext@1.4.0
├─ fined@1.2.0
├─ flagged-respawn@1.0.1
├─ fs-mkdirp-stream@1.0.0
├─ glob-stream@6.1.0
├─ glob-watcher@5.0.5
├─ glogg@1.0.2
├─ gulp-autoprefixer@6.1.0
├─ gulp-clean-css@3.9.0
├─ gulp-cli@2.3.0
├─ gulp-connect@5.7.0
├─ gulp-plumber@1.1.0
├─ gulp-rename@1.2.2
├─ gulp-sass@3.2.1
├─ gulp-util@3.0.8
├─ gulp@4.0.2
├─ has-gulplog@0.1.0
├─ invert-kv@1.0.0
├─ is-relative@1.0.0
├─ is-unc-path@1.0.0
├─ is-valid-glob@1.0.0
├─ json-stable-stringify-without-jsonify@1.0.1
├─ just-debounce@1.0.0
├─ last-run@1.1.1
├─ lazystream@1.0.0
├─ lcid@1.0.0
├─ lead@1.0.0
├─ liftoff@3.1.0
├─ livereload-js@2.4.0
├─ lodash._basecopy@3.0.1
├─ lodash._basetostring@3.0.1
├─ lodash._basevalues@3.0.0
├─ lodash._getnative@3.9.1
├─ lodash._isiterateecall@3.0.9
├─ lodash._reescape@3.0.0
├─ lodash._reevaluate@3.0.0
├─ lodash._root@3.0.1
├─ lodash.clonedeep@4.5.0
├─ lodash.isarguments@3.1.0
├─ lodash.isarray@3.0.4
├─ lodash.keys@3.1.2
├─ lodash.restparam@3.6.1
├─ lodash.template@3.6.2
├─ lodash.templatesettings@3.1.1
├─ map-stream@0.0.7
├─ matchdep@2.0.0
├─ multipipe@0.1.2
├─ mute-stdout@1.0.1
├─ next-tick@1.0.0
├─ object.map@1.0.1
├─ object.reduce@1.0.1
├─ optional@0.1.4
├─ ordered-read-streams@1.0.1
├─ os-locale@1.4.0
├─ papercss@1.7.0
├─ parse-filepath@1.0.2
├─ parse-node-version@1.0.1
├─ path-root-regex@0.1.2
├─ path-root@0.1.1
├─ plugin-error@1.0.1
├─ pretty-hrtime@1.0.3
├─ rechoir@0.6.2
├─ remove-bom-stream@1.2.0
├─ replace-homedir@1.0.0
├─ resolve-options@1.1.0
├─ safe-json-parse@1.0.1
├─ semver-greatest-satisfied-range@1.1.0
├─ stack-trace@0.0.10
├─ stream-exhaust@1.0.2
├─ string_decoder@0.10.31
├─ string-template@0.2.1
├─ sver-compat@1.5.0
├─ through2-filter@3.0.0
├─ time-stamp@1.1.0
├─ tiny-lr@1.1.1
├─ to-absolute-glob@2.0.2
├─ to-through@2.0.0
├─ type@1.2.0
├─ unc-path-regex@0.1.2
├─ undertaker-registry@1.0.1
├─ undertaker@1.2.1
├─ unique-stream@2.3.1
├─ v8flags@3.2.0
├─ vinyl-fs@3.0.3
├─ vinyl-sourcemap@1.1.0
└─ vinyl-sourcemaps-apply@0.2.1
✨  Done in 8.13s.
```



With NPM:
```
> npm install papercss
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated gulp-util@3.0.8: gulp-util is deprecated - replace it, following the guidelines at https://medium.com/gulpjs/gulp-util-ca3b1f9f9ac5

> fsevents@1.2.13 install /Users/vetcove/www/oss/test/node_modules/glob-watcher/node_modules/fsevents
> node install.js

  SOLINK_MODULE(target) Release/.node
  CXX(target) Release/obj.target/fse/fsevents.o
  SOLINK_MODULE(target) Release/fse.node

> papercss@1.7.0 postinstall /Users/vetcove/www/oss/test/node_modules/papercss
> gulp postinstall

[14:21:13] Using gulpfile ~/www/oss/test/node_modules/papercss/gulpfile.js
[14:21:13] Starting 'postinstall'...
[14:21:13] Starting 'sass'...
[14:21:13] Finished 'sass' after 353 ms
[14:21:13] Starting 'minify-css'...
[14:21:13] Finished 'minify-css' after 96 ms
[14:21:13] Finished 'postinstall' after 452 ms
+ papercss@1.7.0
added 197 packages from 140 contributors and audited 1318 packages in 11.289s

41 packages are looking for funding
  run `npm fund` for details

found 1 low severity vulnerability
  run `npm audit fix` to fix them, or `npm audit` for details
```

**Comment by rhyneav** [comment]
This should be fixed with `v1.8.0` as there are no longer any production dependencies!
